File: zs-leetcode-go/ExcelSheetColumnNumber/ExcelSheetColumnNumber.go
package ExcelSheetColumnNumber

func titleToNumber(s string) int {
	length := len(s)
	if length == 0 {
		return 0
	}
	cf := 1
	res := 0
	for index := length-1; index >= 0; index-- {
		res += (int(s[index]-'A')+1)*cf
		cf *= 26
	}
	return res
}

File: sjq-leetcode-C++/excel-sheet-column-number.cpp
class Solution {
public:
    int titleToNumber(string s) {
        int res=0;
        int size = s.size();
        for(int i=0;i<size;i++){
            res += (s[i]-'A'+1) * pow(26,size-i-1);
        }
        return res;
    }
};
